lib/inspec/control_eval_context.rb

```diff
@@ -0,0 +1,139 @@
+# encoding: utf-8
+# author: Dominik Richter
+# author: Christoph Hartmann
+require 'inspec/dsl'
+require 'inspec/dsl_shared'
+
+module Inspec
+  #
+  # ControlEvalContext constructs an anonymous class that control
+  # files will be instance_exec'd against.
+  #
+  # The anonymous class includes the given passed resource_dsl as well
+  # as the basic DSL of the control files (describe, control, title,
+  # etc).
+  #
+  class ControlEvalContext
+    # Create the context for controls. This includes all components of the DSL,
+    # including matchers and resources.
+    #
+    # @param [ResourcesDSL] resources_dsl which has all resources to attach
+    # @return [RuleContext] the inner context of rules
+    def self.rule_context(resources_dsl)
+      require 'rspec/core/dsl'
+      Class.new(Inspec::Rule) do
+        include RSpec::Core::DSL
+        include resources_dsl
+      end
+    end
+
+    # Creates the heart of the control eval context:
+    #
+    # An instantiated object which has all resources registered to it
+    # and exposes them to the a test file.
+    #
+    # @param profile_context [Inspec::ProfileContext]
+    # @param outer_dsl [OuterDSLClass]
+    # @return [ProfileContextClass]
+    #
+    # rubocop:disable Lint/NestedMethodDefinition
+    def self.create(profile_context, resources_dsl) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+      rule_class = rule_context(resources_dsl)
+      profile_context_owner = profile_context
+      profile_id = profile_context.profile_id
+
+      Class.new do
+        include Inspec::DSL
+        include Inspec::DSL::RequireOverride
+        include resources_dsl
+
+        def initialize(backend, conf, dependencies, require_loader)
+          @backend = backend
+          @conf = conf
+          @dependencies = dependencies
+          @require_loader = require_loader
+          @skip_profile = false
+        end
+
+        define_method :title do |arg|
+          profile_context_owner.set_header(:title, arg)
+        end
+
+        def to_s
+          "Profile Context Run #{profile_name}"
```

Comment:
Is this really representing a full profile context? Its only representing one section aka file in a profile